src/Weather/Skysight/Metrics.hpp

```diff
@@ -24,6 +24,91 @@ Copyright_License {
 #ifndef WEATHER_SKYSIGHT_METRICS_HPP
 #define WEATHER_SKYSIGHT_METRICS_HPP
 
+#include "Util/tstring.hpp"
+#include "Time/BrokenDateTime.hpp"
 #include <map>
 
+struct LegendColor {
+  unsigned char Red;
+  unsigned char Green;
+  unsigned char Blue;
+};
+
+struct SkysightMetric {
+  const tstring id;
+  const tstring name;
+  const tstring desc;
+  uint64_t last_update = 0;
+  std::map<float, LegendColor> legend;
+
+public:
+  SkysightMetric(tstring _id, tstring _name, tstring _desc):
+    id(_id), name(_name), desc(_desc) {}
+  SkysightMetric(const SkysightMetric &m):
+    id(m.id), name(m.name), desc(m.desc), last_update(m.last_update),
+    legend(m.legend) {}
+};
+
+struct SkysightActiveMetric {
+  SkysightMetric *metric;
+  double from = 0;
+  double to = 0;
+  double mtime = 0;
+  bool updating = false;
+
+public:
+  SkysightActiveMetric(SkysightMetric *_metric, uint64_t _from,
+		       uint64_t _to, uint64_t _mtime): 
+    metric(_metric), from(_from), to(_to), mtime(_mtime) {}
+  SkysightActiveMetric(const SkysightActiveMetric &m):
+    metric(m.metric), from(m.from), to(m.to), mtime(m.mtime),
+    updating(m.updating) {}
+};
+
+struct DisplayedMetric {
+  SkysightMetric *metric;
+  BrokenDateTime forecast_index;
+  DisplayedMetric() {
+    metric = nullptr;
+  };
+  DisplayedMetric(SkysightMetric *_metric, BrokenDateTime _fc_index):
+    metric(_metric), forecast_index(_fc_index) {};
+  void clear() {
+    metric = nullptr;
+  }
+  bool operator==(const DisplayedMetric &d) {
```

Comment:
These operators look very wrong. `==` compares the id, but `<` and `>` compare the time stamp, while at the same time doing funny things with `IsPlausible`.

Reply:
remove unused functions and operators include this.